### ISSUE TITLE
Lazy loading breaks when the ID field is located in a super-class

### ISSUE THREAD

**Issue by DerNamenlose** [MorphiaOrg/morphia#1643]
**Describe the bug**
When trying to lazy-load a reference which has an inherited ID field (e.g. by having something like an `AbstractEntity` as a base class), the reference never actually gets resolved. Declaring the reference as non-lazy or putting the id field directly in the referenced class, things work just fine.

**To Reproduce**

EntityBase.java:
```java
package morphiaLazyTest;

import dev.morphia.annotations.Entity;
import dev.morphia.annotations.Id;
import dev.morphia.annotations.IdGetter;

@Entity
public abstract class EntityBase {

    @Id
    protected String id;

    @IdGetter
    public String getId()
    {
        return id;
    }
}
```

Entity1.java:
```java
package morphiaLazyTest;

import dev.morphia.annotations.Entity;

@Entity
public class Entity1 extends EntityBase {

    private String name;

    public String getName() {
        return name;
    }

    public void setName(String name) {
        this.name = name;
    }
}
```

Entity2.java:
```java
package morphiaLazyTest;

import dev.morphia.annotations.Entity;
import dev.morphia.annotations.Reference;

@Entity
public class Entity2 extends EntityBase {

    private String anotherName;

    @Reference(idOnly = true, lazy = true)
    private Entity1 reference;

    public String getAnotherName() {
        return anotherName;
    }

    public Entity1 getReference() {
        return reference;
    }

    public void setReference(Entity1 reference) {
        this.reference = reference;
    }

    public void setAnotherName(String anotherName) {
        this.anotherName = anotherName;
    }
}
```

App.java:
```java
package morphiaLazyTest;

import java.util.ArrayList;

import com.mongodb.client.MongoClients;

import dev.morphia.Morphia;
import dev.morphia.query.experimental.filters.Filters;

public class App
{
    public static void main( String[] args )
    {
        var mongoClient = MongoClients.create("<uri>");

        var ds = Morphia.createDatastore(mongoClient, "lazy-test");
        ds.getMapper().map(Entity1.class, Entity2.class, EntityBase.class);

        var entities = ds.find(Entity2.class).iterator().toList();

        System.out.println("E2: "+ entities.get(0));
        System.out.println("Ref: "+entities.get(0).getReference().getId());
    }
}
```

Assuming, that there are two entries in the DB, where an `Entity2` references an `Entity1`, the software outputs:
```
E2: <some object ref>
Ref: null
```

**Expected behavior**

The reference should be loaded and provide the correct data.

**Version info:**

 - Server Version: 5.0.1
 - Driver Version: 4.2.2
 - Morphia Version: 2.2.1

**Comment by DerNamenlose** [comment]
Just adding on, as I've discovered somewhat of a work-around: moving the `getId()` method to the concrete classes (e.g. by requiring the implementation of an abstract method in the base class) is enough to get lazy loading working again.

**Comment by ugljesas** [comment]
Ran into the same issue trying to migrate to new morphia version. Would just like to add that the references do seem to resolve fine, but calling any methods implemented in the base class that access variables declared in the base class is broken. Can we expect a fix for this soon?

**Comment by evanchooly** [comment]
I'm at a conference for a few.more days but I'll take s look when I get home.

**Comment by evanchooly** [comment]
OK.  I'm digging in to this tonight.  I have a feeling its in how the bytebuddy proxy is getting generated.

**Comment by evanchooly** [comment]
fixed by 1eabcff13b44d29fa4af727c16c371a35706cd0d

**Comment by DerNamenlose** [comment]
Thanks for the quick reaction. Looking forward to the 2.2.2.
